lib/kvs/kvs_rparams.c

```diff
@@ -92,6 +95,62 @@ compression_value_algorithm_converter(
     return false;
 }
 
+static merr_t
+compression_value_algorithm_stringify(
+    const struct param_spec *const ps,
+    const void *const              value,
+    char *const                    buf,
+    const size_t                   buf_sz,
+    size_t *const                  needed_sz)
+{
+    int         n;
+    const char *param;
+
+    INVARIANT(ps);
+    INVARIANT(value);
+    INVARIANT(buf);
+
+    const enum vcomp_algorithm algo = *(enum vcomp_algorithm *)value;
+
+    switch (algo) {
+        case VCOMP_ALGO_NONE:
+            param = VCOMP_PARAM_NONE;
+            break;
+        case VCOMP_ALGO_LZ4:
+            param = VCOMP_PARAM_LZ4;
+            break;
+        default:
+            assert(false);
```

Comment:
I prefer switch statements for enums to not have default cases.  This lets the compiler catch missing cases.  I would set `param=NULL`  before the switch, remove the default case, and add `assert(param)` after the switch.   Plus, make in an error case as Greg mentioned.

Reply:
Hmm I see what you are saying. Do you think no default case is better than a default case which is just abort()?

Reply:
I usually do something like this w/ enums:
```
const char *param = NULL;

switch (algo) {
    case VCOMP_ALGO_NONE:
        param = VCOMP_PARAM_NONE;
        break
    case VCOMP_ALGO_LZ4;
        param = VCOMP_PARAM_LZ4;
}

assert(param);
```